### ISSUE TITLE
How make some brackets be completed even in string literals?

### ISSUE THREAD

**Issue by alientechnology** [LucHermitte/lh-brackets#11]
As i see by default brackets not working inside string literals and comments, how to overcome this for some file types and brackets? Let's say i want `{ }` for in string literals for rust.

**Comment by LucHermitte** [comment]
Currently I support the exact contrary with `-context` option.

For instance, if you add in a rust ftplugin:
```
Brackets { } -context=string
Brackets { } -context=string -visual=0 -trigger=µ
```
then `{` will expand only within string context, and `µ` will expand into `{<cursor>}` within strings.

In other words, I have to implement what you're looking for.

I guess I will add a `-context!=` option

**Comment by alientechnology** [comment]
Yeah, I see, but i don't get how to make it explain into `{}<cursor>` within string context, and into `{cursor}` in other contexts? When i add `Brackets { } -context=string` it looks like default `Brackets { }` stops to work. Thanks.

**Comment by LucHermitte** [comment]
OK. I misunderstood what you were looking for. There is a very tricky (and undocumented) way then to achieve it.

```vim
" in case you reload your ft plugin, and if the default global insert-mode mapping for `{` is correct in your context
:silent! iunmap <buffer> {
:call lh#brackets#define_imap('{', [
  \ {'condition' : 'lh#syntax#name_at_mark(".") =~? "string"', 'action': '"{}"'},
  \ {'condition': 1, 'action':maparg('{', 'i')}],
  \ 1)
```
The first condition defines what to do within strings. The second conditions (which expresses the default action) executes the insert mode mapping already registered (here, the one `:Brackets { }` defines.

**Comment by LucHermitte** [comment]
Fixed by 2b14091a6a3fb4918d6aeb5deb3fec22a2ab948c
